### ISSUE TITLE
Variable Products > bulk edit > set sale price from standard price does not work

### ISSUE THREAD

**Issue by Marc-pi** [woocommerce/woocommerce#36802]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

**Bulk Edit**
On Variable Products, set sale price base on standard price does not work
I guess it is because this option was made for single products, so it sets at top level the sale price, not per variation

![image](https://uploads.linear.app/f629fcde-6f9b-48e8-8af3-60eed8823bff/2f907193-99eb-4db9-a37c-0491d6c04b4e/0754b260-4db4-47f1-8a4c-7280b10e4f2b?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2Y2MjlmY2RlLTZmOWItNDhlOC04YWYzLTYwZWVkODgyM2JmZi8yZjkwNzE5My05OWViLTRkYjktYTM3Yy0wNDkxZDZjMDRiNGUvMDc1NGIyNjAtNGRiNC00N2YxLThhNGMtNzI4MGIxMGU0ZjJiIiwiaWF0IjoxNzg4NDI4OTYyLCJleHAiOjE4MTk5OTk1MjJ9.BRjGlKBp_KhuannV919E2NkpNu2f_vdSilByqlZVehU)

\*\*Product Edit (variation tab) \*\*
Such an option does not exists when editing the variable product item
User can only ajust price from an already set sales price
the only way it to set the sales manually on all variations, which is a nightmare

![image](https://uploads.linear.app/f629fcde-6f9b-48e8-8af3-60eed8823bff/5dda81b8-278c-4fb4-8491-0cf002a1eda7/6d436d4a-a4b7-4e11-9481-2ffce436e456?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2Y2MjlmY2RlLTZmOWItNDhlOC04YWYzLTYwZWVkODgyM2JmZi81ZGRhODFiOC0yNzhjLTRmYjQtODQ5MS0wY2YwMDJhMWVkYTcvNmQ0MzZkNGEtYTRiNy00ZTExLTk0ODEtMmZmY2U0MzZlNDU2IiwiaWF0IjoxNzg4NDI4OTYyLCJleHAiOjE4MTk5OTk1MjJ9.G4UGqsORxA3ep83hXAniKlJdlLJAJSXicggvX5CjO70)

### Expected behavior

* Just fix the bulk feature to make it work like for the single product type
  so users can manage his sales for all product types !
* Add such an option within the Variation tab also (set a sales price from standard price)

### Actual behavior

Actually, the feature is not working !
Shops that manage many variable products cannot set their sales, except item per item, but in this case not deduct the sale price from standard price >> users has to edit each variation !!!! Image the time needs when you have hundreds of products
I did not test on other products type (ex: grouped products)

### Steps to reproduce

1. Go to Product > All Products
2. Select a variation product with the checkbox
3. Select Bulk actions > edit to enter bulk mode
4. Product date > sale field : select "Set to regular price decreased by (amount or %)
5. Observe that it has no impact, sales price value is still empty (works like a charm only on single product type)

### WordPress Environment

WP 6.1.1
Woo 7.3.0
Woo Blocks 9.5.0

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](<https://woocommerce.com/storefront/>).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by swatipawarGS** [comment]
Hi @Marc-pi,

Thank you for taking the time to report this bug, we really appreciate your help. We are able to reproduce the issue on our end using the  WooCommerce 7.3.0 version and WordPress 6.1.1 version. 

Please find below the screencast for reference : 

https://user-images.githubusercontent.com/94531721/218395569-65d0c34e-2048-428f-9528-16df3a211958.mp4

Priority of this issue will be assigned shortly after we’d go through a round of prioritization. Please note it may take a couple of days.

**Comment by Marc-pi** [comment]
ok, hope this one can be fixed soon

**Comment by pmcpinto** [comment]
This needs an engineering review before we can make a final call. I've marked it for engineering feedback so someone can look at the reproduction details and the current behavior.

**Comment by vladolaru** [comment]
Hi @Marc-pi, I rechecked this against current trunk. I think this issue is covering two related, but different, workflows.

First, the Products-list bulk edit behavior still matches the report. The UI still exposes `Set to regular price decreased by (fixed amount or %):` for the Sale field: [`html-bulk-edit-product.php`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/admin/views/html-bulk-edit-product.php#L52-L59). The calculation for that option exists, but product-list price updates are gated to simple and external products by default: [`class-wc-admin-post-types.php`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/admin/class-wc-admin-post-types.php#L597-L608), [`class-wc-admin-post-types.php`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/admin/class-wc-admin-post-types.php#L952-L962). That means selecting a variable parent from the Products list does not update its child variations.

I also checked the runtime behavior with a small fixture: a simple product with regular price `100` became sale/active price `90`, while a child variation under a selected variable product stayed at empty sale price and active price `100`.

That Products-list part lines up with the older #28736 / #26763 discussion and the closed #26767 attempt. Updating all child variations from the parent Products-list bulk edit path can become a large synchronous write operation, so I would treat that as the broader #28736 product/UX decision: either design a batched/asynchronous variable-product workflow, or make the Products-list UI clearer that variable product variation prices are not changed there.

The second part of this issue is more scoped and still looks valid: the Variations tab does not have the exact action being requested. It currently offers `Set sale prices`, `Increase sale prices`, and `Decrease sale prices`: [`html-product-data-variations.php`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-variations.php#L101-L108). The existing `Decrease sale prices` action adjusts the `sale_price` field: [`class-wc-ajax.php`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/class-wc-ajax.php#L3104-L3105). The shared price-adjustment helper skips variations where the target field is empty: [`class-wc-ajax.php`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/class-wc-ajax.php#L3144-L3152). So it cannot create a sale price from the regular price when the variation does not already have a sale price.

My recommendation:

1. Keep the Products-list variable-parent behavior tied to #28736 unless we are ready to design the larger batched/asynchronous flow.
2. Improve the Products-list UX if variable prices remain unsupported there, so merchants do not get a successful-looking bulk edit that silently leaves variation prices unchanged.
3. Treat the Variations-tab request as a smaller enhancement: add a new variation bulk action that sets each variation's sale price to its own regular price decreased by a fixed amount or percentage.

That Variations-tab action should be relatively contained. It can reuse the existing variation bulk-action dispatch, fixed/percentage prompt handling, product sync/transient cleanup, and the existing product-list calculation pattern. It would still need a new PHP action/helper because the current `variation_bulk_adjust_price()` helper is based on the target field and intentionally skips blank sale prices. I would add coverage for a variation with regular price `100` and no sale price, then verify the new action sets sale price to `90` for a `10%` decrease.

**Comment by dinhtungdu** [comment]
Assigning my self to add the new action to variation bulk action.

**Comment by dinhtungdu** [comment]
I’m closing in favor of https://github.com/woocommerce/woocommerce/issues/28736 as the convo is happening there.
